app/ide-desktop/lib/dashboard/src/hooks/debounceCallbackHooks.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file
+ *
+ * This file contains the `useDebouncedCallback` hook which is used to debounce a callback function.
+ */
+import * as React from 'react'
+
+import * as callbackHooks from './eventCallbackHooks'
+import * as unmountEffect from './unmountEffectHooks'
+
+/**
+ * Wrap a callback into debounce function
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function useDebouncedCallback<Fn extends (...args: any[]) => any>(
+  callback: Fn,
+  deps: React.DependencyList,
+  delay: number,
+  maxWait = 0
+): DebouncedFunction<Fn> {
+  const callbackEvent = callbackHooks.useEventCallback(callback)
```

Comment:
i don't think `callbackEvent` makes sense as the name for this

Reply:
idk tbh. i'd probably do something like `stableCallback` (or perhaps `latestCallback`) (or make the original input `callbackRaw`, although in this case that would affect the funtion signature). it's not ideal but less misleading at least?